onnxruntime/core/providers/cpu/generator/constant_of_shape.cc

```diff
@@ -105,6 +105,18 @@ inline void FilloutOutput(T value, Tensor* output_tensor) {
   std::fill(out.begin(), out.end(), value);
 }
 
+ConstantOfShapeBase::ConstantOfShapeBase(const OpKernelInfo& info){
+  TensorProto t_proto;
+  if (info.GetAttr<TensorProto>("value", &t_proto).IsOK()) {
+    ORT_ENFORCE(t_proto.dims_size() == 1, "Must have a single dimension");
+    ORT_ENFORCE(t_proto.dims()[0] == 1, "Must have a single dimension of 1");
```

Comment:
If the input is scalar there would be zero dims and one value. Spec says "one element tensor" so I think a scalar tensor with no dims is valid input.

Reply:
Scalar tensor still needs dims, right? [https://github.com/onnx/onnx/blob/7a112a6f930aac375da5a38d744ccda93e37632c/onnx/onnx-ml.proto#L308](https://github.com/onnx/onnx/blob/7a112a6f930aac375da5a38d744ccda93e37632c/onnx/onnx-ml.proto#L308)  

---
In reply to: [291098621](https://github.com/microsoft/onnxruntime/pull/1168#discussion_r291098621) [](ancestors = 291098621)